docs/aws/audit/redshiftmonitoring/rules/unused_redshift_reserved_nodes_remediation.mdx

```diff
@@ -1,6 +1,148 @@
+</Accordion>
+
+</AccordionGroup>
+</Tab>
+<Tab title='Cause'>
+### Check Cause
+<AccordionGroup>
+<Accordion title='Using Console' defaultOpen='true'>
+1. Sign in to the AWS Management Console.
+2. Navigate to the Redshift dashboard. From the navigation pane, choose "Clusters".
+3. In the Clusters dashboard, you will see a list of all your Redshift clusters. Check the status of each cluster. If a cluster's status is "available", it means it's currently in use. If the status is "creating", "deleting", "rebooting", "renaming", or "resizing", it means the cluster is not currently in use.
+4. To check if a Reserved Node is unused, navigate to the "Reserved Nodes" section in the navigation pane. Here, you will see a list of all your Reserved Nodes. Check the "Node Type", "State", and "Duration" of each Reserved Node. If the State is "active" but the Node is not associated with any cluster, it means the Reserved Node is unused.
+</Accordion>
+
+<Accordion title='Using CLI'>
+1. First, you need to install and configure AWS CLI on your local machine. You can download it from the official AWS website and configure it using the "aws configure" command. You will be prompted to provide your AWS Access Key ID, Secret Access Key, Default region name, and Default output format.
+
+2. Once the AWS CLI is set up, you can list all the Redshift Reserved Nodes using the following command:
+
+   ```
+   aws redshift describe-reserved-nodes
+   ```
+
+   This command will return a JSON output with details of all the reserved nodes.
+
+3. To check if a Reserved Node is unused, you need to check the "State" attribute in the output. If the "State" attribute is "available", it means that the Reserved Node is not currently being used.
+
+4. To automate this process, you can use a Python script with the Boto3 library (AWS SDK for Python) to list all the Reserved Nodes and check their states. Here is a simple script:
+
+   ```python
+   import boto3
+
+   # Create a Redshift client
+   redshift = boto3.client('redshift')
+
+   # List all reserved nodes
+   reserved_nodes = redshift.describe_reserved_nodes()
+
+   # Check if any reserved node is unused
+   for node in reserved_nodes['ReservedNodes']:
+       if node['State'] == 'available':
+           print(f"Unused Reserved Node: {node['ReservedNodeId']}")
+   ```
+
+   This script will print the IDs of all unused Reserved Nodes.
+</Accordion>
+
+<Accordion title='Using Python'>
+1. Install and configure AWS SDK for Python (Boto3) on your local system. Boto3 allows you to directly create, update, and delete AWS services from your Python scripts.
+
+2. Import the necessary libraries and establish a session using your AWS credentials. Here's a sample script to get started:
+
+```python
+import boto3
+
+session = boto3.Session(
+    aws_access_key_id='YOUR_ACCESS_KEY',
+    aws_secret_access_key='YOUR_SECRET_KEY',
```

Comment:
<b>Secret:</b> Secret Keyword <br/>
                    <b><a href="https://console.cloudanix.com/code?t=secrets" target='_blank'>Click here for more details</a></b>